packages/react/ds/styles.css

```diff
@@ -17,29 +17,61 @@
   opacity: 0;
 }
 
-/* Header Styles */
+@layer components {
 
-#GovieHeader {
+  /* Header Styles */
 
-  #MenuContainer:has(#SearchTrigger:checked) ~ #SearchContainer {
-    height: var(--gieds-space-40);
+  #GovieHeader {
+
+    #MenuContainer:has(#SearchTrigger:checked) ~ #SearchContainer {
+      height: var(--gieds-space-40);
+    }
+    
+    #SearchTrigger:checked ~ .search-icon {
+      display: none; 
+    }
+    
+    #SearchTrigger:checked ~ .close-icon {
+      display: block; 
+    }
+    
+    #MenuContainer:has(#MobileMenuTrigger:checked) ~ #HeaderMenuContainer {
+      transform: translateX(0);
+    }
+    #MenuContainer:has(#MobileMenuTrigger:checked) ~ #HeaderOverlayContainer {
+      display: block;
+      position: fixed;
+    }
```

Comment:
Sneak feature? 😂

Reply:
not mine this one :D

Reply:
That's my doing 😄 are we saying we should refactor it ?

Reply:
Nothing to see here.